indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -162,12 +162,15 @@ default int getPriority()
 
   /**
    * Asks a task to arrange for its "run" method to exit promptly. Tasks that take too long to stop gracefully will be
-   * terminated with extreme prejudice.
+   * terminated with extreme prejudice. Note that this method can be called at any time while {@link #run} is called.
+   * Its implementations should handle potential concurreny issues properly.
    *
-   * If the task has some resources to clean up on exit, e.g., sub tasks of parallel indexing task
-   * or Hadoop job of Hadoop indexing task, those resource cleanups should be done in this method.
+   * If the task has some resources to clean up on exit, e.g., sub tasks of parallel indexing task or Hadoop job of
+   * Hadoop indexing task, those resource cleanups should be done in this method.
    *
    * @param taskConfig TaskConfig for this task
+   *
+   * @see org.apache.druid.indexing.worker.http.WorkerResource#doShutdown(String)
```

Comment:
Could mention that this shutdown APi will cause stopGracefully to be called either through lifecycle stop (ForkingTaskRunner on MM -> process shutdown -> SingleTaskBackgroundRunner on Peon) or directly from an HTTP thread (ThreadingTaskRunner for Indexer)

Reply:
Added some description about it.